src/index.js

```diff
@@ -1,10 +1,167 @@
+  changeMiddleSegment(allCityDetails)
+  getScrollWidth()
+  icons.forEach((ic) => ic.addEventListener('click', selectIcon))
+  leftArrow.addEventListener('click', () => { scrollItems('left') })
+  rightArrow.addEventListener('click', () => { scrollItems('right') })
+  document.getElementById('select-icon').addEventListener('change', getCardCount)
+
+  /**
+   *  This function will get the scrollwidth and clientwidth of the itemscontainer to display the arrow
+   */
+  function getScrollWidth () {
+    const middleContainerWidth = document.getElementsByClassName('items-container')[0].clientWidth
+    const scrollContainerWidth = document.getElementsByClassName('items-container')[0].scrollWidth
+    if (middleContainerWidth >= scrollContainerWidth) {
+      document.getElementsByClassName('left-arr')[0].style.display = 'none'
+      document.getElementsByClassName('right-arr')[0].style.display = 'none'
+      document.getElementsByClassName('items-container')[0].style.justifyContent = 'space-around'
+    } else {
+      document.getElementsByClassName('left-arr')[0].style.display = 'block'
+      document.getElementsByClassName('right-arr')[0].style.display = 'block'
+    }
+  }
+
+  /**
+   * This function will get the value from spinner  and update the card numbers
+   */
+  function getCardCount () {
+    cardCount = document.getElementById('select-icon').value
+    changeMiddleSegment(middleSegmentCards)
+    getScrollWidth()
+  }
+
+  /**
+   * This function will scroll the container when arrow is clicked
+   * @param {string} arrow It is used to identify the arrow
+   */
+  function scrollItems (arrow) {
+    const cardWidth = document.querySelector('.items-scroll').getBoundingClientRect().width
+    const scrollContainer = document.querySelector('.items-container')
+    if (arrow === 'right') scrollContainer.scrollLeft += cardWidth
+    else scrollContainer.scrollLeft -= cardWidth
+  }
+
+  /**
+   *  This function sorts the objects
+   * @param {object} data  all city details
+   *  @returns {object} returns the sorted city objects
+   */
+  function sortCityOptions (data) {
+    const sortedKeys = Object.keys(data).sort()
+    const result = { }
+    sortedKeys.forEach((key) => { result[key] = data[key] })
+    console.log(result)
+    return result
+  }
+
+  /**
+   *   This function selects the icon and add style. It calls the filter function
+   */
+  function selectIcon () {
+    if (selectedIcon !== ' ') { document.getElementById(selectedIcon).classList.remove('ic-click') }
+    selectedIcon = this.id
+    document.getElementById(this.id).classList.add('ic-click')
+    filterMiddleSegment(this.id)
+  }
+
+  /**
+   *  This function will filter the cities based on selected icons
+   * @param {string} iconId icon id for identification
+   */
+  function filterMiddleSegment (iconId) {
+    let filterObject = {}
+    document.getElementsByClassName('items-container')[0].innerHTML = ' '
+    Object.values(allCityDetails).map((data) => {
```

Comment:
Why not store it as array during the initialization itself?

Reply:
Instead of `map` operator, can we try to make use of `filter` operator?
It'll filter directly based on the condition we provide and won't need to create a list on our own again like this 
![image](https://github.com/Abishek666/abishek_WebTips_Assignments/assets/142198666/5f37428f-bef8-4f55-b4a1-a73018ebd149)